textattack/search_methods/particle_swarm_optimization.py

```diff
@@ -24,252 +25,313 @@ class ParticleSwarmOptimization(SearchMethod):
+        max_turn_retries (:obj:`bool`, optional): Maximum number of movement retries if new position after turning fails to pass the constraints.
+            Applied only when `post_movement_check` is set to `True`.
+            Setting it to 0 means we immediately take the old position as the new position upon failure.
     """
 
     def __init__(
-        self, pop_size=60, max_iters=20,
+        self, pop_size=60, max_iters=20, post_turn_check=True, max_turn_retries=20
     ):
         self.max_iters = max_iters
         self.pop_size = pop_size
-        self.search_over = False
+        self.post_turn_check = post_turn_check
+        self.max_turn_retries = 20
 
-        self.Omega_1 = 0.8
-        self.Omega_2 = 0.2
+        self._search_over = False
+        self.omega_1 = 0.8
+        self.omega_2 = 0.2
         self.C1_origin = 0.8
         self.C2_origin = 0.2
         self.V_max = 3.0
 
-    def _generate_population(self, x_orig, neighbors_list, neighbors_len):
-        h_score, w_list = self._gen_h_score(x_orig, neighbors_len, neighbors_list)
-        return [self._mutate(x_orig, h_score, w_list) for _ in range(self.pop_size)]
+    def _perturb(self, pop_member, original_result):
+        """Perturb `pop_member` in-place.
 
-    def _mutate(self, x_cur, w_select_probs, w_list):
-        rand_idx = np.random.choice(len(w_select_probs), 1, p=w_select_probs)[0]
-        return x_cur.replace_word_at_index(rand_idx, w_list[rand_idx])
-
-    def _gen_h_score(self, x, neighbors_len, neighbors_list):
+        Replaces a word at a random in `pop_member` with replacement word that maximizes increase in score.
+        Args:
+            pop_member (PopulationMember): The population member being perturbed.
+            original_result (GoalFunctionResult): Result of original sample being attacked
+        Returns:
+            `True` if perturbation occured. `False` if not.
+        """
+        # TODO: Below is very slow and is the main cause behind memory build up + slowness
+        best_neighbors, prob_list = self._get_best_neighbors(
+            pop_member.result, original_result
+        )
+        random_result = np.random.choice(best_neighbors, 1, p=prob_list)[0]
 
-        w_list = []
-        prob_list = []
-        for i, orig_w in enumerate(x.words):
-            if neighbors_len[i] == 0:
-                w_list.append(orig_w)
-                prob_list.append(0)
-                continue
-            p, w = self._gen_most_change(x, i, neighbors_list[i])
-            w_list.append(w)
-            prob_list.append(p)
+        if random_result == pop_member.result:
+            return False
+        else:
+            pop_member.attacked_text = random_result.attacked_text
+            pop_member.result = random_result
+            return True
 
-        prob_list = self._norm(prob_list)
+    def _equal(self, a, b):
```

Comment:
That is from the original code base. Is there an alternative name you think is better?

Reply:
I don't really understand what it's doing, but I feel like I would think an `_equal` function which takes 2 inputs would return a boolean determining whether they are equal, rather than a number based on if they are equal.

Reply:
<img width="295" alt="Screen Shot 2020-07-25 at 6 19 44 PM" src="https://user-images.githubusercontent.com/32072203/88453594-6ce29080-cea3-11ea-82c7-cc7cc4bfdf26.png">

This is the actual function it's supposed to be... I'm not sure why return max velocity, however. 
I agree that `_equal` isn't fitting, but, I don't think it's a huge issue. If you still want to rename it to something else, I'll be more than happy to change it.